### PR TITLE
chore(dependencies): bump minimum torch from 2.2.1 to 2.7

### PR DESCRIPTION
**Pull request by CarolinePascal**
## Type / Scope

- **Type**: Chore
- **Scope**: ffmpeg, video encoding/decoding

## Summary / Motivation

This PR aims at making `ffmpeg` installation instructions clearer depending in the users' installed versions of `pytoch` and `torchcodec`.

## Related issues

- Fixes / Closes: # (if any)
- Related: # (if any)

## What changed

- Installation: no more restrictions on `ffmpeg` version (+ new instructions for the `torchcodec` now supported system-wide `ffmpeg` installation)
- Dependencies: `torch>=2.7` and `torchcodec>=0.3.0`

## How was this tested (or how to run locally)

TBD :
- [ ] Installation using system-wide `ffmpeg` (`uv` based installation)
- [ ] Dataset recording (`av` test)
- [ ] Training (`torchcodec` test)
- [ ] Inference(both)

## Checklist (required before merge)

- [ ] Linting/formatting run (`pre-commit run -a`)
- [ ] All tests pass locally (`pytest`)
- [ ] Documentation updated
- [ ] CI is green

## Reviewer notes

- Anything the reviewer should focus on (performance, edge-cases, specific files) or general notes.
- Anyone in the community is free to review the PR.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -71,9 +71,9 @@ dependencies = [
     "cmake>=3.29.0.1,<4.2.0",
     "packaging>=24.2,<26.0",
 
-    "torch>=2.2.1,<2.11.0",
-    "torchcodec>=0.2.1,<0.11.0; sys_platform != 'win32' and (sys_platform != 'linux' or (platform_machine != 'aarch64' and platform_machine != 'arm64' and platform_machine != 'armv7l')) and (sys_platform != 'darwin' or platform_machine != 'x86_64')",
-    "torchvision>=0.21.0,<0.26.0",
+    "torch>=2.7,<2.10.0",
```
Comment (imstevenpmwork) [inline pyproject.toml]: no edit ceil bound

Comment (CarolinePascal) [reply pyproject.toml]: Done in [599f5ac](https://github.com/huggingface/lerobot/pull/3156/commits/599f5ac987f466ec174a7c0c7b68f909e547fb6f)

File: docs/source/installation.mdx [hunk 2]
```diff
@@ -38,24 +38,39 @@ conda activate lerobot
...
+> For the full compatibility matrix, see the [TorchCodec installation table](https://github.com/meta-pytorch/torchcodec?tab=readme-ov-file#installing-cpu-only-torchcodec).
+
+If your platform supports TorchCodec, install `ffmpeg` using one of the methods below:
+
+<!-- prettier-ignore-start -->
+
+<hfoptions id="install_ffmpeg">
+<hfoption id="conda (any PyTorch version)">
+
+Install `ffmpeg` in your conda environment. This works with **all PyTorch versions** and is **required for PyTorch < 2.10**:
```
Comment (imstevenpmwork) [inline docs/source/installation.mdx:84]: Did you mean 2.11 maybe?

File: docs/source/installation.mdx [hunk 3]
```diff
@@ -38,24 +38,39 @@ conda activate lerobot
...
 
 ```bash
 conda install ffmpeg -c conda-forge
-ffmpeg -version  # ffmpeg 8.X is not yet supported !
 ```
 
 > [!TIP]
-> This usually installs `ffmpeg 7.X` for your platform compiled with the `libsvtav1` encoder. If `libsvtav1` is not supported (check supported encoders with `ffmpeg -encoders`), you can:
+> This usually installs `ffmpeg 7.X` with the `libsvtav1` encoder. If you run into issues (e.g. `libsvtav1` missing — check with `ffmpeg -encoders` — or a version mismatch with `torchcodec`), you can:
```
Comment (imstevenpmwork) [inline docs/source/installation.mdx]: This would now prioritize 8.0, we should change the note version

File: docs/source/installation.mdx (hunk 2 above)
Comment (CarolinePascal) [reply docs/source/installation.mdx:84]: Support starts from version 2.10, so conda is still needed for versions < 2.10 !

File: docs/source/installation.mdx (hunk 3 above)
Comment (CarolinePascal) [reply docs/source/installation.mdx]: Done in 3a5ed576e70e4fefdf3045fcf0df151fc9d83745
